packages/api/src/middleware/validation/sale.js

```diff
@@ -17,11 +17,11 @@ import RevenueTypeModel from '../../models/revenueTypeModel.js';
 async function validateSale(req, res, next) {
   // TODO replace upsertGraph
   const { crop_variety_sale, revenue_type_id } = req.body;
-  // TODO: implement properly once LF-3595 is complete
-  const cropSaleRevenueType = await RevenueTypeModel.query()
-    .where('revenue_name', 'Crop Sale')
-    .first();
-  const isCropRevenue = revenue_type_id === cropSaleRevenueType.revenue_type_id;
+
+  const revenueType = await RevenueTypeModel.query().findById(revenue_type_id);
```

Comment:
Not your code but it is weird that this is middleware. Its kind of similar to a schema check but also not really. 

Don't feel obligated to take action but it seems like this check should be in the controller for the same trx reasons we have been discussing lately.

Reply:
For a minute this line crashed my db -- then I cleared cache and it worked normal. 

So revenue_type_id wasn't passed just noting in case anyone else has it. 
![Screenshot 2023-09-29 at 2 09 54 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/4667fda6-6068-4f67-8037-4368c05dab60)

But technically a savvy api user could crash our db by not supplying it.

Reply:
Oh gnarly, `findById()` sucks 😆 I'll use `.where()` and `.first()` instead then

Reply:
a simple check beforehand is probs fine too -- I am guessing it is a performant function

Reply:
Ok. I guess since it needs a check anyway might as well put it first. We use the `.where()` pattern a lot more but I imagine you're right about performance.